content/en/articles/js/react/unit-testing.md

```diff
@@ -0,0 +1,212 @@
+
+Let's do this with the following (dump) component:
+
+```js
+// web_modules/Picture/index.js
+
+import React from "react"
+import { PropTypes } from "react"
+
+const Component = ({
+  img,
+  title,
+  Loader,
+  Title,
+}) => (
+  <div>
+    {
+      (!img || !img.src) && Loader &&
+      <Loader />
+    }
+    {
+      img && img.src &&
+      <img src={ img.src } alt={ img.alt }/>
+    }
+    {
+      title && Title &&
+      <Title text={ title } />
+    }
+  </div>
+)
+
+Component.propTypes = {
+  img: PropTypes.object,
+  title: PropTypes.string,
+  Loader: PropTypes.func.isRequired,
+  Title: PropTypes.func.isRequired,
+}
+
+Component.displayName = "Picture"
+
+export default Component
+```
+
+This component displays an image with a title component.
+If the image data are not ready yet, it can display a loader component.
+
+Now let's write some simple test for it. For the example we will use
+[tape](https://medium.com/javascript-scene/why-i-use-tape-instead-of-mocha-so-should-you-6aa105d8eaf4)
+with the help of [tape-jsx-equals](https://github.com/atabel/tape-jsx-equals),
+but you will find
+[all kind of flavors on npm](https://www.npmjs.com/search?q=expect+jsx).
+
+```js
+// web_modules/Picture/__tests__/index.js
+
+import tape from "tape"
+import addAssertions from "extend-tape"
+import jsxEquals from "tape-jsx-equals"
+const test = addAssertions(tape, { jsxEquals })
+
+import React from "react"
+import { createRenderer } from "react-addons-test-utils"
+
+import Picture from ".."
+
+// fixtures (empty and stateless react components)
+const Loader = () => {}
+const Title = () => {}
+
+test("PageContainer is properly rendered", (t) => {
+  const renderer = createRenderer()
+
+  renderer.render(
+    <Picture
+      Loader={ Loader }
+      Title={ Title }
+    />
+  )
+  t.jsxEquals(
+    renderer.getRenderOutput(),
+    <div>
+      <Loader />
+    </div>,
+    "can render a Loader component if no image data are passed"
+  )
+
+  renderer.render(
+    <Picture
+      Loader={ Loader }
+      Title={ Title }
+      img={ {
+        src: "SRC",
+        alt: "ALT",
+      } }
+    />
+  )
+  t.jsxEquals(
+    renderer.getRenderOutput(),
+    <div>
+      <img src="SRC" alt="ALT" />
+    </div>,
+    "should render an image if data are passed"
+  )
+
+  renderer.render(
+    <Picture
+      Loader={ Loader }
+      Title={ Title }
+      img={ {
+        src: "SRC",
+        alt: "ALT",
+      } }
+      title={ "TITLE" }
+    />
+  )
+  t.jsxEquals(
+    renderer.getRenderOutput(),
+    <div>
+      <img src="SRC" alt="ALT" />
+      <Title text="TITLE" />
+    </div>,
+    "can render a Title if data are passed"
+  )
+
+  t.end()
+})
+```
+
+These tests are the minimum coverage to ensure you don't break anything when you work
+on your component.
+
+As you can see, tests here are pretty easy to write & straightforward.  
+**The interesting part is that you don't compare using strings.
```

Comment:
I would say the interesting part is that you are testing the output you implemented (JSX), not the resulting React.createElement calls nor doing lots of calls to element.props....